R/geom-segment.r

```diff
@@ -42,6 +43,19 @@
 #'     arrow = arrow(length = unit(0.1,"cm"))) +
 #'   borders("state")
 #'
+#' # Use lineend and linejoin to change the style of the segments
+#' df2 <- expand.grid(lineend = c('round', 'butt', 'square'),
+#'                    linejoin = c('round', 'mitre', 'bevel'), stringsAsFactors = FALSE)
+#' segments <- lapply(seq_len(nrow(df2)), function(i) {
```

Comment:
This use of `lapply()` will confuse most readers. Can you please generate with a simple helper function and copy and paste?